src/common/cluster-store/cluster-store.injectable.ts

```diff
@@ -36,6 +37,7 @@ const clusterStoreInjectable = getInjectable({
     persistStateToConfig: di.inject(persistStateToConfigInjectionToken),
     enlistMessageChannelListener: di.inject(enlistMessageChannelListenerInjectionToken),
     shouldDisableSyncInListener: di.inject(shouldBaseStoreDisableSyncInIpcListenerInjectionToken),
+    sendMessageToChannel: di.inject(sendMessageToChannelInjectionToken),
```

Comment:
There's obviously an existing pattern for this but it seems painful that this change has to be done for every store.

Reply:
Yes I agree this is quite painfull.

I think a better solution would be for `fromStore` and `toJson` to be configuration params to `BaseStore` instead of `abstract` methods.

That way we could have one `createStoreInjectable` that is referenced instead of all of the dependencies being mandatory in many places.

Namely, composition over inheritance.